src/cluster/managers/awaiters/Awaiter.ts

```diff
@@ -1,54 +1,60 @@
 import { PromiseCompletionSource } from '@blargbot/core/PromiseCompletionSource';
 
 export class Awaiter<T> {
-    private readonly timeout: NodeJS.Timeout;
-    private readonly pcs: PromiseCompletionSource<T | undefined>;
+    readonly #timeout: NodeJS.Timeout;
+    readonly #pcs: PromiseCompletionSource<T | undefined>;
+    readonly #poolIds: ReadonlySet<string>;
+    readonly #pools: Record<string, Array<Awaiter<T>> | undefined>;
+    readonly #check: (item: T) => Awaitable<boolean>;
 
     public constructor(
-        private readonly poolIds: ReadonlySet<string>,
-        private readonly pools: Record<string, Array<Awaiter<T>> | undefined>,
-        private readonly check: (item: T) => Awaitable<boolean>,
+        poolIds: ReadonlySet<string>,
+        pools: Record<string, Array<Awaiter<T>> | undefined>,
+        check: (item: T) => Awaitable<boolean>,
         timeout: number
     ) {
-        this.pcs = new PromiseCompletionSource<T | undefined>();
+        this.#pcs = new PromiseCompletionSource<T | undefined>();
+        this.#poolIds = poolIds;
+        this.#pools = pools;
+        this.#check = check;
 
         for (const poolId of poolIds)
             (pools[poolId] ??= []).push(this);
 
-        this.timeout = setTimeout(() => this.cancel(), timeout);
+        this.#timeout = setTimeout(() => this.cancel(), timeout);
     }
 
     public async wait(): Promise<T | undefined> {
-        return await this.pcs.promise;
+        return await this.#pcs.promise;
     }
```

Comment:
not related to the private fields change, but wouldn't this be changed to something like:
```ts
    public wait(): Promise<T | undefined> {
        return this.#pcs.promise;
    }
```

Reply:
it could, but in general im not a fan of returning promises. It hides methods from the call stack in case of an exception.